papermill/abs.py

```diff
@@ -4,7 +4,7 @@
 
 from six.moves import urllib
 
-from azure.storage.blob import BlockBlobService
```

Comment:
Likely this is because we have the dependency pinned back for the earlier version in the requirements.txt. We need to make this file backwards compatible with regards to imports and function calls. Unfortunately given the changed needed this looks like it'll require trying the new path and falling back to the old on import exception.

Reply:
Is it ok to change the pin to be for version >=12.0.0? Or are you saying that we can't upgrade to the new version? (Sorry I'm still pretty new to python, I don't really understand the build process for it)

Reply:
It'd be ideal if we supported both  `<=2.1.0` and `>2.1.0`. If it's easier for you, just solve for the most recent version (leave the dependency as just `azure-storage-blob`) and I can make the backwards compatibility fix later.

Reply:
Alright, I'll leave the backwards compatibility to you ;) Thanks!